src/shared/components/Filter.tsx

```diff
@@ -25,31 +27,39 @@ export type Props<T> = OwnProps<T> & StateProps
 
 const INEXACT_PATH = /\w+\[\]/g
 const EMPTY_ARRAY_BRACKETS = /\[\]?\./
+
 /**
  * Filters a list using a searchTerm and searchKeys where
  *  searchKeys is an array of strings represents either an
  *  exact or inexact path to a property value(s):
  *  "user.name" (exact) or "authors[].name" (inexact)
  *
  */
+export default class FilterList<T extends Resource> extends PureComponent<
+  Props<T>
+> {
+  private collator: Intl.Collator
+
+  public constructor(props) {
+    super(props)
+    this.collator = new Intl.Collator('en-us', {numeric: true})
+  }
 
-export default class FilterList<T> extends PureComponent<Props<T>> {
   public render() {
     return this.props.children(this.sorted)
   }
 
   private get sorted(): T[] {
-    return sortBy<T>(this.filtered, [
-      (item: T) => {
-        const value = get(item, this.props.sortByKey)
-
-        if (!!value && typeof value === 'string') {
-          return value.toLocaleLowerCase()
-        }
+    return this.filtered.sort((item1, item2) => {
+      if (this.props.sortByKey) {
+        return this.collator.compare(
+          item1[this.props.sortByKey],
+          item2[this.props.sortByKey]
+        )
+      }
 
-        return value
-      },
-    ])
+      return this.collator.compare(item1.name, item2.name)
```

Comment:
this is where the magic is. [`Intl.Collator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) will sort naturally if passed an option of `numeric: true`. Previously used lodash `orderBy` sorted lexicographically. I took this opportunity to use a native function, rather than one vended by an external library.